api_fhir_r4/converters/activityDefinitionConverter.py

```diff
@@ -73,18 +80,48 @@ def get_imis_obj_by_fhir_reference(cls, reference, errors=None):
         return DbManagerUtils.get_object_or_none(Service, code=imis_activity_definition_code)
 
     @classmethod
-    def build_fhir_identifiers(cls, fhir_activity_definition, imis_activity_definition):
+    def build_fhir_identifiers(cls, fhir_activity_definition, imis_activity_definition, reference_type):
         identifiers = []
-        cls.build_all_identifiers(identifiers, imis_activity_definition)
+        cls.build_fhir_uuid_identifier(identifiers, imis_activity_definition)
+        cls.build_fhir_code_identifier(identifiers, imis_activity_definition)
+        if reference_type == ReferenceConverterMixin.DB_ID_REFERENCE_TYPE:
+            cls.build_fhir_id_identifier(identifiers, imis_activity_definition)
         fhir_activity_definition.identifier = identifiers
 
+    @classmethod
+    def build_imis_price(cls, imis_activity_definition, fhir_activity_definition, reference_type):
+        extension_base_url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/unit-price"
+        extension = cls.get_fhir_extension_by_url(fhir_activity_definition.extension,
+                                                  extension_base_url)
+
+        if extension and hasattr(extension, "valueMoney"):
+            imis_activity_definition.price = extension.valueMoney.value
+
+    @classmethod
+    def build_imis_frequency(cls, imis_activity_definition, fhir_activity_definition, errors):
+        value = None
+        try:
```

Comment:
I like the solution with the try, except, finally block for assigning values / adding information to the errors list. 
It's quite repetitive in every method. Do you think we could make it more generic, e.g. with creating a wrapper  implementing this logic in parent class and apply it for every build_imis_* method? 
It could also be reused in other converters for fhir -> IMIS mapping, and therefore improve error handling. @delcroip what do you think about it?

Reply:
it sound possible https://stackoverflow.com/a/27500036/3533113

but it might be tricky because the assignment is on  "fhir_activity_definition.timingTiming.repeat.period" but you check if fhir_activity_definition have timingTiming attributes, I think a recursive function might be required

Reply:
I'm not sure we need recursive approach. I think `hasattr(fhir_activity_definition, "timingTiming")` is only for checking if timing was provided in the payload. If it is there, we can assume that it's correctly formatted and contains repeat and period. However if there are issues with nested elements then message is added to the errors list.

Reply:
That was the idea.

Reply:
This is a nice to have. https://openimis.atlassian.net/browse/OE0-54